### PR TITLE
suit: Allow installing Nordic elements from external memory

### PR DESCRIPTION
**Pull request by ahasztag**
Also, the possibility of adding external flash to the recovery application was added

test_suit_dfu: PR-418

### CODE REVIEW

**Review by SylwesterKonczyk — APPROVED** [review]

File: samples/suit/smp_transfer/boards/nrf54h20dk_nrf54h20_cpuapp_extflash.overlay [hunk 1]
```diff
@@ -22,3 +22,24 @@
...
+	};
+
+	cpuapp_slot0_partition: partition@ba000 {
+		reg = <0xba000 DT_SIZE_K(1124)>;
+	};
+};
+
+&cpuapp_rw_partitions {
+	/delete-node/ partition@100000;
+	/delete-node/ dfu_partition;
```
Comment (tomchy) [inline samples/suit/smp_transfer/boards/nrf54h20dk_nrf54h20_cpuapp_extflash.overlay:41]: I have a feeling that line 40 and 41 deletes the same partition... is that true?

File: cmake/sysbuild/suit.cmake [hunk 2]
```diff
@@ -469,6 +469,46 @@ function(suit_create_package)
 
   sysbuild_get(DEFAULT_BINARY_DIR IMAGE ${DEFAULT_IMAGE} VAR APPLICATION_BINARY_DIR CACHE)
 
+  if(SB_CONFIG_SUIT_ENVELOPE_NORDIC_TOP_EXTRACT_PAYLOADS_TO_CACHE)
+
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:473]:
```suggestion
```

File: sysbuild/Kconfig.suit [hunk 3]
```diff
@@ -47,12 +47,37 @@ config SUIT_ENVELOPE_ROOT_ARTIFACT_NAME
...
+	default y if SUIT_BUILD_FLASH_COMPANION
+	help
+		When this option is enabled, the build system extracts payloads from
+		the nordic_top envelope and stores them into a cache partition file.
+		Note, if this option is enabled, the installation of the Nordic components
+		must be orchestrated by the APP_ROOT envelope.
+		The root envelope must make use of the In-Place Updateable Components (IPUC)
+		feature - one of the MEM components must be marked as IPUC by setting its
+		size to 0 in the suit-candidate-verification sequence - so that its
+		memory can be used as a mirror for the update of the Nordic elements.
```
Comment (nordicjm) [inline sysbuild/Kconfig.suit:73]: help indent is 1x tab followed by 2x spaces

File: cmake/sysbuild/suit.cmake (hunk 2 above)
Comment (ahasztag) [reply cmake/sysbuild/suit.cmake]: Fixed

File: cmake/sysbuild/suit_utilities.cmake [hunk 4]
```diff
@@ -110,6 +110,22 @@ function(suit_create_cache_partition args output_file partition_num recovery)
   endif()
 endfunction()
 
+function(suit_create_nordic_cache_partition args output_file)
+
```
[the remark this answers is not in the document]
Comment (ahasztag) [reply cmake/sysbuild/suit_utilities.cmake]: Fixed

File: sysbuild/Kconfig.suit (hunk 3 above)
Comment (ahasztag) [reply sysbuild/Kconfig.suit]: Fixed

**Review by FrancescoSer — APPROVED** [review]

**Review by tomchy — APPROVED** [review]

**Review by adsz-nordic — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**